server/src/controllers/song.controllers.ts

```diff
@@ -0,0 +1,106 @@
+import { Request, RequestHandler, Response } from 'express';
+import { Song }  from '../models/song.model';
+import { SongDocument } from '../types/song.types';
+
+const sendResponse = (res: Response, statusCode: number, payload: SongDocument[] | SongDocument | string ) => {
+  return res.status(statusCode).json(payload);
+};
+
+const createSong: RequestHandler = async (req: Request, res: Response): Promise<SongDocument | string> => {
+  const { ...toCreate } = req.body;
+
+  if (Object.keys(toCreate).length > 0) {
+    try {
+      const data = await Song.create(toCreate);
+
+      if (data) {
+        sendResponse(res, 200, data);
+      } else {
+        sendResponse(res, 404, 'Song not created');
+      }
+    } catch (error: any) {
+      sendResponse(res, 500, error.message);
+    }
+  } else {
+    sendResponse(res, 400, 'Missing required fields' );
+  }
+  return '';
+};
+
+const getSong: RequestHandler = async (req: Request, res: Response): Promise<SongDocument[] | SongDocument | string> => {
+    const { id: songId } = req.params;
+
+    if (songId) {
+      try {
+        const data = await Song.findOne({ _id: songId }).exec();
+  
+        if (data) {
+          sendResponse(res, 200, data);
+        } else {
+          sendResponse(res, 404, 'Song not found');
+        }
+      } catch (error: any) {
+        sendResponse(res, 500, error.message );
+      }
+    } else {
+      try {
+        const data = await Song.find().exec();
```

Comment:
I think u need to add params `{isDeleted: false}` inside the find, so it'll only show the ones that are not soft deleted. same thing with the find specific id, and other find calls.